### PR TITLE
Update validate to check for missing components

### PR DESCRIPTION
**Pull request by bourdakos1**
Note: This is completely untested
 
Fixes https://github.com/elyra-ai/elyra/issues/2192
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/PipelineController/index.ts [hunk 1]
```diff
@@ -238,45 +238,6 @@ class PipelineController extends CanvasController {
...
-          invalidNodeError: `"${node.op}" is an unsupported node type`,
-        },
-        description: undefined,
-        image: image,
-      },
-      pipelineID
-    );
-    this.setNodeLabel(node.id, "unsupported node", pipelineID);
-  }
-
```
Comment (ajbozarth) [inline packages/pipeline-editor/src/PipelineController/index.ts:279]: was this just an unused function that was meant to do this validation?

File: packages/pipeline-services/src/validation/index.ts [hunk 2]
```diff
@@ -62,6 +62,16 @@ export function getNodeProblems(pipeline: any, nodeDefinitions: any) {
...
+      problems.push({
+        message: `The component '${node.op}' cannot be found.`,
+        path: ["nodes", n],
+        info: {
+          type: "missingComponent",
+          pipelineID: pipeline.id,
+          nodeID: node.id,
+          op: node.op,
+        },
+      });
```
Comment (ajbozarth) [inline packages/pipeline-services/src/validation/index.ts:74]: so we just added a "problem" when hitting the edge case we already caught and ignored?

**Review by ajbozarth — COMMENTED** [review]
A couple clarifying questions while I'm building to test locally.

Was this a partially implemented validation that we just weren't leveraging before?

**Comment by ajbozarth** [comment]
Ran this locally and it does render the error outline, but I also noticed that for this use case the tooltip doesn't render either (also on master). So the user won't know why it failed

Screen Shot 2021-10-05 at 12 46 08 PM

File: packages/pipeline-services/src/validation/index.ts (hunk 2 above)
Comment (bourdakos1) [reply packages/pipeline-services/src/validation/index.ts:74]: Correct

File: packages/pipeline-editor/src/PipelineController/index.ts (hunk 1 above)
Comment (bourdakos1) [reply packages/pipeline-editor/src/PipelineController/index.ts:279]: This was originally intended to catch nodes that were only available to kfp/airflow when switching between pipeline type. Since it’s not possible to switch pipeline types this turned into dead code. This piece of code turns the icon for the node into a question mark and changes the label to “invalid node” which is poor user experience, because you can’t tell what the node was intended to be

**Comment by bourdakos1** [comment]
> Ran this locally and it does render the error outline, but I also noticed that for this use case the tooltip doesn't render either (also on master). So the user won't know why it failed
> 
> Screen Shot 2021-10-05 at 12 46 08 PM

Oh yea we need to set the error message on the node

**Comment by ajbozarth** [comment]
> Oh yea we need to set the error message on the node
 
do you want ti fix that here or merge this and fix it later?

File: packages/pipeline-editor/src/PipelineController/index.test.ts [hunk 3]
```diff
@@ -841,87 +841,6 @@ describe("properties", () => {
...
-
-    expect(setNodeLabel).toHaveBeenCalledTimes(1);
-    expect(setNodeLabel).toHaveBeenCalledWith(
-      "node-to-find",
-      "unsupported node",
-      "pipeline1"
-    );
-  });
-
-  it("doesn't set style for supernodes", () => {
```
Comment (ajbozarth) [inline packages/pipeline-editor/src/PipelineController/index.test.ts:892]: Why are you deleting this test?

File: packages/pipeline-services/src/validation/index.test.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ describe("validate", () => {
     expect(problems).toHaveLength(0);
   });
 
-  it("should return an empty array if there are no problems", () => {
+  it("should find missing components", () => {
```
Comment (ajbozarth) [inline packages/pipeline-services/src/validation/index.test.ts:26]: Doesn't this remove the no errors test case, should we have both?

File: packages/pipeline-editor/src/PipelineController/index.test.ts (hunk 3 above)
Comment (ajbozarth) [reply packages/pipeline-editor/src/PipelineController/index.test.ts:892]: nvm I see now that this is a subtest for the function you removed

File: packages/pipeline-services/src/validation/index.test.ts (hunk 4 above)
Comment (bourdakos1) [reply packages/pipeline-services/src/validation/index.test.ts:26]: we have other tests that cover no errors, this one was a bit redundant
